zds/utils/forms.py

```diff
@@ -108,8 +109,10 @@ def validate_string_list(self, string_list):
         :return: ``True`` if ``v`` is fully valid, ``False`` if at least one error appears. See ``self.errors``
         to get all internationalized error.
         """
+        string_list = list(filter(lambda s: s.strip(), string_list))
         self.__clean = list(filter(self.validate_length, string_list))
         self.__clean = list(filter(self.validate_utf8mb4, self.__clean))
+        self.__clean = list(filter(self.validate_no_empty_slug, self.__clean))
```

Comment:
tant qu'à faire, virer tous les `list(…)` pour en laisser qu'un. En appelant `list`, ça matérialise le générateur en tant que vraie liste, et on peut très bien chaîner les `filter` en matérialisant qu'une seule fois à la fin.

Reply:
yep, je regarde pour celui-là.

Reply:
Ah, si seulement on pouvais `self.__clean |> filter(…) |> filter(…) |> filter(…)`… :upside_down_face:

Reply:
à la java en somme ?

Reply:
C’est plutôt à la [Elixir](https://elixirschool.com/en/lessons/basics/pipe-operator/) ou à la [F#](http://theburningmonk.com/2011/09/fsharp-pipe-forward-and-pipe-backward/) :upside_down_face: 

(Mais je ne connais pas Java au delà de Java 7 parce que je suis un vieux)